flutter_app/lib/modules/custom_log_functions.dart

```diff
@@ -0,0 +1,9 @@
+import 'dart:developer';
+
+void logDroneMode(int? currentMode, String? currentModeName) {
```

Comment:
I think this is making it to complicated, I see where you're coming from but I think it's better to keep it in the widget